### PR TITLE
various provide-trader-eori fixes

### PR DESCRIPTION
**Pull request by RobertLaverick**
fixed issue where supplied eori wasn't sent to backend
fixed issue where special characters weren't checked for correctly
preserved default error case (though using duplicate messaging)
improved default and special character case messaging

### CODE REVIEW

File: conf/messages.cy [hunk 1]
```diff
@@ -865,7 +865,8 @@ provideTraderEori.error.required=Enter the trader''s EORI number
 provideTraderEori.error.tooLong=EORI number is too long
 provideTraderEori.error.tooShort=EORI number is not long enough
 provideTraderEori.error.notGB=Enter EORI number beginning with GB
-provideTraderEori.error.general=EORI country code can only be GB followed by 12 digits, like GB123456123456
+provideTraderEori.error.specialCharacters=Enter an EORI number in the correct format, like GB123456789010
```
Comment (payal-bonde) [inline conf/messages.cy:868]: Error message set earlier was correct, can we not have same here as well  - EORI country code can only be GB followed by 12 digits, like GB123456123456.

File: conf/messages.cy [hunk 2]
```diff
@@ -865,7 +865,8 @@ provideTraderEori.error.required=Enter the trader''s EORI number
...
+provideTraderEori.error.default=Enter an EORI number in the correct format, like GB123456789010
```
Comment (payal-bonde) [inline conf/messages.cy:869]: Error message set earlier was correct, can we not have same here as well  - EORI country code can only be GB followed by 12 digits, like GB123456123456.

Comment (RobertLaverick) [reply conf/messages.cy:869]: See above - that change was intentional and the new message is from Amanda

**Review by payal-bonde — APPROVED** [review]
Approved

**Review by C-Thomp — APPROVED** [review]
🥇

**Pull request merged**